src/engraving/dom/staff.cpp

```diff
@@ -104,6 +104,10 @@ staff_idx_t Staff::idx() const
 
 void Staff::triggerLayout() const
 {
+    if (staffType() && staffType()->isJianpu()) {
+        score()->style().set(Sid::MusicalSymbolFont, "Jianpu");
```

Comment:
@cbjeukendrup I agree with you! 

I think we need to add a new style parameter something like `JianpuFont`
And get the font implementation by `Sid` or staff type 

that is, we need to do something like this:
* generalise EngravingFont (and maybe add type to EngravingFontsProvider::addFont, see enum Font::Type) 
* remove Score::m_engravingFont
* change `Score:engravingFont()` to two methods: `Score:engravingFont(Sid )` and  `Score:engravingFont(const StaffType* )`
* try compile, fix all code where use `Score:engravingFont` (change `LayoutConfiguration::engravingFont`)

Reply:
I don't think we should have `engravingFont(StaffType*)’. It's not that different staff types should use different fonts. It's that the different staff type uses different symbols at different codepoints, that are not part of the SMuFL spec. That's why we can't use SmuFL fonts for that so we need to combine a SMuFL font with a Jianpu font preferably in one structure (like EngravingFont) so that we can reuse our SymId and symBbox logic etc. The question is how exactly that should work though. Especially since would ideally want the user to be able to choose any text font as their Jianpu font, but for text fonts there is of course no metadata file with bbox info, so we need to compute that info ourselves.

Reply:
Actually, since Jianpu characters _are_ just text, we should maybe just use plain strings to represent the symbols and FontMetrics to get the metrics. (Of course, we can use named constants for each Jianpu character instead of writing string literals everywhere.)
How do we feel about that idea?

Reply:
I like your idea.
I'm thinking about how this task is related to the display of Tablatures, it also has its own font...
I also think that this task is rather infrastructural and has little to do with the current PR (that is, we need to improve the font system as general). Therefore, I would prefer that this PR be done as desired, the just working..., and then the system fonts need to be improved separately.